crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -69,76 +72,97 @@ pub async fn sys_validation_workflow(
         // that we are meant to be holding but aren't.
         // If we are not holding them they will be added to our incoming ops.
         let incoming_dht_ops_sender =
             IncomingDhtOpSender::new(workspace.vault.clone(), sys_validation_trigger.clone());
-
-        let outcome = validate_op(
-            &op,
-            workspace,
-            network.clone(),
-            &conductor_api,
-            Some(incoming_dht_ops_sender),
-        )
-        .await?;
-
-        match outcome {
-            Outcome::Accepted => {
-                workspace
-                    .put_validation_limbo(op_hash, ValidationLimboStatus::SysValidated)
-                    .await?;
-            }
-            Outcome::SkipAppValidation => {
-                workspace
-                    .put_integration_limbo(op_hash, ValidationStatus::Valid)
-                    .await?;
-            }
-            Outcome::AwaitingOpDep(missing_dep) => {
-                // TODO: Try and get this dependency to add to limbo
-                //
-                // I actually can't see how we can do this because there's no
-                // way to get an DhtOpHash without either having the op or the full
-                // header. We have neither that's why where here.
-                //
-                // We need to be holding the dependency because
-                // we were meant to get a StoreElement or StoreEntry or
-                // RegisterAgentActivity or RegisterAddLink.
-                let status = ValidationLimboStatus::AwaitingSysDeps(missing_dep);
-                workspace.put_validation_limbo(op_hash, status).await?;
-            }
-            Outcome::MissingDhtDep => {
-                // TODO: Not sure what missing dht dep is. Check if we need this.
-                workspace
-                    .put_validation_limbo(op_hash, ValidationLimboStatus::Pending)
-                    .await?;
-            }
-            Outcome::Rejected => {
-                workspace
-                    .put_integration_limbo(op_hash, ValidationStatus::Rejected)
-                    .await?;
-            }
+        let network = network.clone();
+        let conductor_api = conductor_api.clone();
+        let workspace = workspace.clone();
+        async move {
+            let conductor_api = conductor_api.clone();
+            let (op, op_hash) = so.into_inner();
+            let r = validate_op(
+                &op,
+                &(*workspace),
+                network,
+                &(*conductor_api),
+                Some(incoming_dht_ops_sender),
+            )
+            .await;
+            r.map(|o| (op_hash, o))
         }
+    });
+    use futures::stream::StreamExt;
+    let mut iter = futures::stream::iter(iter)
+        .buffer_unordered(NUM_CONCURRENT_OPS)
+        .ready_chunks(NUM_CONCURRENT_OPS);
+
+    while let Some(chunk) = iter.next().await {
+        workspace
+            .vault
+            .async_commit(move |mut txn| {
```

Comment:
if we've already got these in batches, should we do them in the same transaction?

Reply:
hah! nevermind, that is what this is doing : )